pkg/controller/installation/core_controller.go

```diff
@@ -1696,7 +1706,75 @@ func (r *ReconcileInstallation) setDefaultsOnFelixConfiguration(install *operato
 			}
 		}
 	}
-	return updated
+
+	// If BPF is enabled, but not set on FelixConfiguration, do so here. This could happen when an older
+	// version of operator is replaced by the new one. Older versions of the operator used an
+	// environment variable to enable BPF, but we no longer do so. In order to prevent disruption
+	// when the environment variable is removed by the render code of the new operator, make sure
+	// FelixConfiguration has the correct value set.
+
+	// If calico-node daemonset exists, we need to check the ENV VAR and set FelixConfiguration accordingly.
+	// Otherwise, just move on.
+	ds := &appsv1.DaemonSet{}
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: common.CalicoNamespace, Name: common.NodeDaemonSetName}, ds)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			reqLogger.Error(err, "An error occurred when getting the Daemonset resource")
+			return false, err
+		}
+	} else {
+		bpfEnabledOnDaemonsetWithEnvVar, err := bpfEnabledOnDaemonsetWithEnvVar(ds)
+		if err != nil {
+			reqLogger.Error(err, "An error occurred when querying the Daemonset resource")
+			return false, err
+		} else if bpfEnabledOnDaemonsetWithEnvVar && !bpfEnabledOnFelixConfig(fc) {
+			err = setBPFEnabledOnFelixConfiguration(fc, true)
+			if err != nil {
+				reqLogger.Error(err, "Unable to enable eBPF data plane")
+				return false, err
+			} else {
+				updated = true
+			}
+		}
+	}
+
+	return updated, nil
+}
+
+// setBPFUpdatesOnFelixConfiguration will take the passed in fc and update any BPF properties needed
+// based on the install config and the daemonset.
+func (r *ReconcileInstallation) setBPFUpdatesOnFelixConfiguration(ctx context.Context, install *operator.Installation, fc *crdv1.FelixConfiguration, reqLogger logr.Logger) (bool, error) {
+	updated := false
+
+	bpfEnabledOnInstall := install.Spec.BPFEnabled()
+	if bpfEnabledOnInstall {
+		ds := &appsv1.DaemonSet{}
+		err := r.client.Get(ctx, types.NamespacedName{Namespace: common.CalicoNamespace, Name: common.NodeDaemonSetName}, ds)
+		if err != nil {
+			return false, err
+		}
+		if !bpfEnabledOnFelixConfig(fc) && isRolloutCompleteWithBPFVolumes(ds) {
+			err := setBPFEnabledOnFelixConfiguration(fc, bpfEnabledOnInstall)
+			if err != nil {
+				reqLogger.Error(err, "Unable to enable eBPF data plane")
+				return false, err
+			} else {
+				updated = true
+			}
+		}
+	} else {
+		if fc.Spec.BPFEnabled == nil || *fc.Spec.BPFEnabled {
+			err := setBPFEnabledOnFelixConfiguration(fc, bpfEnabledOnInstall)
+			if err != nil {
+				reqLogger.Error(err, "Unable to enable eBPF data plane")
```

Comment:
Is this correct? `bpfEnabledOnInstall == false` so arent's we trying to explicitly disable?

Reply:
Good catch! Raised [PR](https://github.com/tigera/operator/pull/3307) to fix it. 
@caseydavenport PTAL. Thanks.